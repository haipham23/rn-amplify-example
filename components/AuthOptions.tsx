import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import { Button, Text } from 'native-base';

interface Props {
  currentFormState: string,
  showRegister(): void,
  showLogin(): void,
  showForgetPassword(): void,
  showResendEmail(): void,
}

const AuthOptions = ({
  currentFormState,
  showRegister,
  showLogin,
  showForgetPassword,
  showResendEmail
}: Props) => (
  <View style={styles.content}>
    {
      currentFormState !== 'register' && (
        <Button transparent onPress={showRegister}>
          <Text>Register</Text>
        </Button>
      )
    }
    {
      currentFormState !== 'login' && (
        <Button transparent onPress={showLogin}>
          <Text>Login</Text>
        </Button>
      )
    }
    {
      currentFormState !== 'forgetPassword' && (
        <Button transparent onPress={showForgetPassword}>
          <Text>Forget your password?</Text>
        </Button>
      )
    }
    {
      currentFormState !== 'resendEmail' && (
        <Button transparent onPress={showResendEmail}>
          <Text>Resend confirmation code.</Text>
        </Button>
      )
    }
  </View>
);

const styles = StyleSheet.create({
  content: {
    padding: 10
  }
});

export default AuthOptions;
