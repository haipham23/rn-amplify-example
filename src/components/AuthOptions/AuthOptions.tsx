import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import { Button, Text } from 'native-base';

interface Props {
  currentFormState: string,
  showRegister(): void,
  showLogin(): void,
  showForgetPassword(): void
}

const AuthOptions = ({
  currentFormState,
  showRegister,
  showLogin,
  showForgetPassword
}: Props) => (
  <View style={styles.content}>
    {
      currentFormState !== 'register' && (
        <Button transparent onPress={showRegister}>
          <Text>Register</Text>
        </Button>
      )
    }
    {
      currentFormState !== 'login' && (
        <Button transparent onPress={showLogin}>
          <Text>Login</Text>
        </Button>
      )
    }
    {
      currentFormState !== 'forgetPassword' && (
        <Button transparent onPress={showForgetPassword}>
          <Text>Forget your password?</Text>
        </Button>
      )
    }
  </View>
);

const styles = StyleSheet.create({
  content: {
    padding: 10
  }
});

export default AuthOptions;
