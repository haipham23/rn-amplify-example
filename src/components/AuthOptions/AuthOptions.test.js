import * as React from 'react';
import renderer from 'react-test-renderer';

import AuthOptions from './AuthOptions';

describe('<AuthOptions />', () => {
  const showRegister = jest.fn();
  const showLogin = jest.fn();
  const showForgetPassword = jest.fn();

  it('should render', () => {
    const tree = renderer.create(
      <AuthOptions
        currentFormState="register"
        showRegister={showRegister}
        showLogin={showLogin}
        showForgetPassword={showForgetPassword}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
