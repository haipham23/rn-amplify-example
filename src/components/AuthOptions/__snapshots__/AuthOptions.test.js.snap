// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthOptions /> should render 1`] = `
<View
  style={
    Object {
      "padding": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 0,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#007aff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Login
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 0,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#007aff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Forget your password?
    </Text>
  </View>
</View>
`;
